webview/src/shared/components/dragDrop/DragDropContainer.tsx

```diff
@@ -72,33 +72,44 @@ export const DragDropContainer: React.FC<DragDropContainerProps> = ({
   // eslint-disable-next-line sonarjs/cognitive-complexity
 }) => {
   const [draggedOverId, setDraggedOverId] = useState('')
+  const [hoveringSomething, setHoveringSomething] = useState(false)
```

Comment:
[Q] what is the difference between `hoveringSomething` and `isHovering`, why do we need both?

Reply:
They are working together. And it's because of:
```
const handleDragLeave = () => {
    isHovering.current = false
    hoveringTimeout.current = window.setTimeout(() => {
      if (!isHovering.current) {
        setHoveringSomething(false)
      }
    }, 500)
  }
  ```
`onDragLeave` is a little too reactive and changing the state every time is a lot. Not only that but there is also a race condition with `onDragEnter` of other elements.